src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -581,4 +583,36 @@ private static void assertNotNull(AssertionInfo info, Class<?> actual) {
   private static void classParameterIsNotNull(Class<?> clazz) {
     requireNonNull(clazz, "The class to compare actual with should not be null");
   }
+
+  /**
+   * Verifies that the actual {@code Class} has the given {@code packagename}.
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param packageName the package that must be declared in the class.
+   */
+  public void assertHasPackage(AssertionInfo info, Class<?> actual, String packageName) {
+    assertNotNull(info, actual);
+    requireNonNull(packageName, shouldNotBeNull("package").create());
+    Package actualPackage = actual.getPackage();
+
+    if (actualPackage == null || !actualPackage.getName().equals(packageName)) {
+      throw failures.failure(info, shouldHavePackage(actual, packageName));
+    }
+  }
+
+  /**
+   * Verifies that the actual {@code Class} has the given {@code Package}.
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param aPackage the package that must be declared in the class.
+   */
+  public void assertHasPackage(AssertionInfo info, Class<?> actual, Package aPackage) {
```

Comment:
```suggestion
  public void assertHasPackage(AssertionInfo info, Class<?> actual, Package expected) {
```

Reply:
On second thought, now that I'm reviewing this modification in the IDE, I don't like it very much because it's not consistent with the other method in the class. For the String variant, it's named "packageName" which is explicative and makes sense, and the related error message is constructed like `shouldNotBeNull("packageName")`. But here changing it to "expected" loses some information and it's not coherent with the other methods in the class, where this semantic is not used. I would change it back to "aPackage", "pkg", or I'm accepting better name suggestions.